crates/wash-lib/src/build/provider.rs

```diff
@@ -0,0 +1,132 @@
+use std::io::ErrorKind;
+use std::path::PathBuf;
+use std::process;
+
+use anyhow::{anyhow, bail, Context, Result};
+use nkeys::KeyPairType;
+use tracing::warn;
+
+use crate::build::SignConfig;
+use crate::cli::par::{create_provider_archive, detect_arch, ParCreateArgs};
+use crate::cli::{extract_keypair, OutputKind};
+use crate::parser::{CommonConfig, LanguageConfig, ProviderConfig};
+
+/// Build a capability provider for the current machine's architecture
+/// and operating system using provided configuration.
+pub(crate) async fn build_provider(
+    provider_config: &ProviderConfig,
+    language_config: &LanguageConfig,
+    common_config: &CommonConfig,
+    signing_config: Option<&SignConfig>,
+) -> Result<PathBuf> {
+    let LanguageConfig::Rust(rust_config) = language_config else {
+        bail!("Unsupported language for provider: {:?}", language_config)
+    };
+    let mut command = match rust_config.cargo_path.as_ref() {
+        Some(path) => process::Command::new(path),
+        None => process::Command::new("cargo"),
+    };
+
+    // Change directory into the project directory
+    std::env::set_current_dir(&common_config.path)?;
+
+    // Build for a specified target if provided, or the default rust target
+    let build_args = if let Some(override_target) = &provider_config.target {
+        vec!["build", "--release", "--target", override_target]
+    } else {
+        vec!["build", "--release"]
+    };
+    let result = command.args(build_args).status().map_err(|e| {
+        if e.kind() == ErrorKind::NotFound {
+            anyhow!("{:?} command is not found", command.get_program())
+        } else {
+            anyhow!(e)
+        }
+    })?;
+
+    if !result.success() {
+        bail!("Compiling provider failed: {}", result.to_string())
```

Comment:
Super nit: I don't think you need to `to_string()` it here as it should implement `Display`